File: GotadeOuroDefault.Web/Modules/Produto/Marca/MarcaForm.cs
using Serenity.ComponentModel;

namespace GotadeOuroDefault.Produto.Forms;

[FormScript("Produto.Marca")]
[BasedOnRow(typeof(MarcaRow), CheckNames = true)]
public class MarcaForm
{
    public string Nome { get; set; }
    public bool MarcaNacional { get; set; }
}

File: GotadeOuroDefault.Web/Modules/ServerTypes/Produto/MarcaForm.ts
import { StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface MarcaForm {
    Nome: StringEditor;
    MarcaNacional: BooleanEditor;
}

export class MarcaForm extends PrefixedContext {
    static formKey = 'Produto.Marca';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MarcaForm.init)  {
            MarcaForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(MarcaForm, [
                'Nome', w0,
                'MarcaNacional', w1
            ]);
        }
    }
}